### PR TITLE
Fix payload decoding and error handling.

### PR DESCRIPTION
**Pull request by sri2k1us**
- This will address some PR comments given to me by @slr71

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -104,8 +116,9 @@ func EmailRequestHandler(w http.ResponseWriter, r *http.Request, emailSettings E
 		emailReq, payloadMap, err := parseRequestBody(r)
 		if err != nil {
 			logcabin.Error.Println(err)
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(err.Error()))
+			//w.WriteHeader(getErrorResponseCode(err))
+			//w.Write([]byte(err.Error()))
+			JSONError(w, r, (string(err.Error())), getErrorResponseCode(err))
```
Comment (slr71) [inline main.go:121]: `err.Error()` returns a string, so you can feel free to remove the type cast.

File: main.go [hunk 2]
```diff
@@ -114,8 +127,9 @@ func EmailRequestHandler(w http.ResponseWriter, r *http.Request, emailSettings E
 			formattedMsg, err := FormatMessage(emailReq, payloadMap, deSettings)
 			if err != nil {
 				logcabin.Error.Println(err)
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
+				//	w.WriteHeader(http.StatusInternalServerError)
+				//	w.Write([]byte(err.Error()))
+				JSONError(w, r, (string(err.Error())), getErrorResponseCode(err))
```
Comment (slr71) [inline main.go:132]: You can remove the type cast from here too.

**Review by slr71 — APPROVED** [review]
Looks good to me! :shipit: whenever you're ready.

**Comment by sri2k1us** [comment]
Thanks for the review.

**Pull request merged**